plugins/catalog-backend-module-unprocessed/src/UnprocessedEntitiesModule.ts

```diff
@@ -0,0 +1,132 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  HydratedRefreshState,
+  RefreshState,
+  UnprocessedEntitiesRequest,
+  UnprocessedEntitiesResponse,
+} from './types';
+import { Knex } from 'knex';
+import { HttpRouterService } from '@backstage/backend-plugin-api';
+import Router from 'express-promise-router';
+import { getBearerTokenFromAuthorizationHeader } from '@backstage/plugin-auth-node';
+
+/**
+ * Module providing Unprocessed Entities API endpoints
+ *
+ * @public
+ */
+export class UnprocessedEntitesModule {
+  private readonly moduleRouter;
+
+  constructor(
+    private readonly database: Knex,
+    private readonly router: HttpRouterService,
+  ) {
+    this.moduleRouter = Router();
+    this.router.use(this.moduleRouter);
+  }
+
+  private async unprocessed(
+    request: UnprocessedEntitiesRequest,
+  ): Promise<UnprocessedEntitiesResponse> {
+    if (request.reason === 'pending') {
+      return {
+        type: 'pending',
+        entities: await this.pending(request.owner),
+      };
+    }
+    return {
+      type: 'failed',
+      entities: await this.failed(request.owner),
+    };
+  }
+
+  private hydrateRefreshState(r: RefreshState): HydratedRefreshState {
+    return {
+      ...r,
+      unprocessed_entity: JSON.parse(r.unprocessed_entity),
+      ...(r.processed_entity && {
+        processed_entity: JSON.parse(r.processed_entity),
+      }),
+      ...(r.errors && { errors: JSON.parse(r.errors) }),
+      ...(r.cache && { cache: JSON.parse(r.cache) }),
+    };
+  }
+
+  private async pending(owner?: string): Promise<HydratedRefreshState[]> {
+    const res = (
+      await this.database('refresh_state.*')
+        .from('refresh_state')
+        .leftJoin(
+          'final_entities',
+          'final_entities.entity_id',
+          'refresh_state.entity_id',
+        )
+        .whereNull('final_entities.entity_id')
+    ).map(this.hydrateRefreshState);
+    if (owner) {
+      return res.filter(r => r.unprocessed_entity.spec?.owner === owner);
+    }
+
+    return res;
+  }
+
+  private async failed(owner?: string): Promise<HydratedRefreshState[]> {
+    const res = (
+      await this.database('refresh_state.*')
+        .from('refresh_state')
+        .rightJoin(
+          'final_entities',
+          'final_entities.entity_id',
+          'refresh_state.entity_id',
+        )
+        .whereNull('final_entities.final_entity')
+    ).map(this.hydrateRefreshState);
+    if (owner) {
+      return res.filter(r => r.unprocessed_entity.spec?.owner === owner);
+    }
+
+    return res;
+  }
+
+  registerRoutes() {
```

Comment:
One refactor I would probably suggest here, is to make this function called `getRouter` and then you can return a `Router()` instance in here and then use it in both the module by using `router.use(module.getRouter())` and the same for the old system too. Makes it a little cleaner, and it means less dependencies injected into the contructor.